Lib/importlib/_bootstrap.py

```diff
@@ -263,7 +263,13 @@ def _requires_builtin_wrapper(self, fullname):
 def _requires_frozen(fxn):
     """Decorator to verify the named module is frozen."""
     def _requires_frozen_wrapper(self, fullname):
-        if not _imp.is_frozen(fullname):
+        if not _imp.is_frozen(fullname) or (
+            # bpo-45272: os.path is not a concrete module, but an importable
+            #            attribute of 'os'. _imp.is_frozen('os.path') returns
+            #            False but it is frozen, so we check the loader here.
+            fullname in sys.modules
+            and sys.modules[fullname].__spec__.loader is FrozenImporter
+        ):
```

Comment:
Hmm, I'm pretty sure we don't need to add the special case here any more.  I don't remember why I had to freeze os.path in the first place, other than that a test was failing.  It must have been something that was fixed elsewhere, since the test suite passes now.  So go ahead and drop this change.

```suggestion
        if not _imp.is_frozen(fullname):
```